Bug during run in a TF Cloud

### Terraform Version
```
0.12.16
```

### Terraform Configuration Files
_I am not sure what excatly is relevant, because running locally the TF works OK._

### Crash Output
```
Failed running command "cd /terraform/terraform; envdir /env terraform show -json /terraform/terraform.tfplan" (exit 2)
Output: 

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: 

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
panic: can't use ElementIterator on null value

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.ElementIterator(0x2310080, 0xc000b9f9e0, 0x0, 0x0, 0x1ce0ea0, 0x1)
	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/zclconf/go-cty@v1.1.0/cty/value_ops.go:922 +0xd5
github.com/hashicorp/terraform/command/jsonstate.marshalAttributeValues(0x2310080, 0xc000b9f9e0, 0x0, 0x0, 0xc000700fe0, 0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonstate/state.go:109 +0xc4
github.com/hashicorp/terraform/command/jsonstate.marshalResources(0xc00110c510, 0xc0001f2f70, 0x0, 0x0, 0xc00080fe98, 0x7fd1beeb0d10, 0xc0010f0ea0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonstate/state.go:325 +0x254
github.com/hashicorp/terraform/command/jsonstate.marshalRootModule(0xc000b9e398, 0xc0001f2f70, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonstate/state.go:193 +0x107
github.com/hashicorp/terraform/command/jsonstate.(*state).marshalStateValues(0xc000ab6a50, 0xc000b9e398, 0xc0001f2f70, 0x0, 0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonstate/state.go:159 +0x139
github.com/hashicorp/terraform/command/jsonstate.Marshal(0xc00110c300, 0xc0001f2f70, 0x0, 0x0, 0x0, 0x2e, 0x23c4)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonstate/state.go:139 +0x9e
github.com/hashicorp/terraform/command/jsonplan.Marshal(0xc0005dc410, 0xc0002439d0, 0xc00110c300, 0xc0001f2f70, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/jsonplan/plan.go:123 +0x75d
github.com/hashicorp/terraform/command.(*ShowCommand).Run(0xc0002f3380, 0xc00003a0a0, 0x2, 0x2, 0xc00044c4c0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/show.go:147 +0x929
github.com/mitchellh/cli.(*CLI).Run(0xc000460280, 0xc000460280, 0xc000531d90, 0x1)
	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f1
main.wrappedMain(0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:238 +0xc34
main.realMain(0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:102 +0xb4
main.main()
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:38 +0x3b
```

### Expected Behavior
Running TF

### Actual Behavior
TERRAFORM CRASH
```
Failed running command "cd /terraform/terraform; envdir /env terraform show -json /terraform/terraform.tfplan" (exit 2)
```

### Steps to Reproduce
1. Migrating to cloud
2. Running plan in the cloud


### References
Not known so far

--- a/command/jsonplan/values.go
+++ b/command/jsonplan/values.go
@@ -26,7 +26,7 @@ type stateValues struct {
 type attributeValues map[string]interface{}
 
 func marshalAttributeValues(value cty.Value, schema *configschema.Block) attributeValues {
-	if value == cty.NilVal {
+	if value == cty.NilVal || value.IsNull() {
 		return nil
 	}
 	ret := make(attributeValues)

--- a/command/jsonplan/values_test.go
+++ b/command/jsonplan/values_test.go
@@ -31,6 +31,18 @@ func TestMarshalAttributeValues(t *testing.T) {
 			nil,
 		},
 		{
+			cty.NullVal(cty.String),
+			&configschema.Block{
+				Attributes: map[string]*configschema.Attribute{
+					"foo": {
+						Type:     cty.String,
+						Optional: true,
+					},
+				},
+			},
+			nil,
+		},
+		{
 			cty.ObjectVal(map[string]cty.Value{
 				"foo": cty.StringVal("bar"),
 			}),

--- a/command/jsonstate/state.go
+++ b/command/jsonstate/state.go
@@ -101,9 +101,10 @@ type resource struct {
 type attributeValues map[string]interface{}
 
 func marshalAttributeValues(value cty.Value, schema *configschema.Block) attributeValues {
-	if value == cty.NilVal {
+	if value == cty.NilVal || value.IsNull() {
 		return nil
 	}
+
 	ret := make(attributeValues)
 
 	it := value.ElementIterator()

--- a/command/jsonstate/state_test.go
+++ b/command/jsonstate/state_test.go
@@ -92,6 +92,18 @@ func TestMarshalAttributeValues(t *testing.T) {
 			nil,
 		},
 		{
+			cty.NullVal(cty.String),
+			&configschema.Block{
+				Attributes: map[string]*configschema.Attribute{
+					"foo": {
+						Type:     cty.String,
+						Optional: true,
+					},
+				},
+			},
+			nil,
+		},
+		{
 			cty.ObjectVal(map[string]cty.Value{
 				"foo": cty.StringVal("bar"),
 			}),

